model/proxy.yaml

```diff
@@ -0,0 +1,20 @@
+groups:
+  - id: proxy
```

Comment:
How can we determine whether something is a proxy or actually the final client/server?

Should we instead name this `peer` or `remote` or `next_hop` or `direct_peer` or `socket_peer` or something like that and define it to be just that?

Reply:
I would prefer the name `next_hop` (or any other of my suggestions). I think that is easier to understand than "proxy", esp. in situations where there might be multiple intermediaries/proxies.

Reply:
I think it would then be valid to set both but the next_hop is more low-level technical information so should probably be "more optional". I see next_hop (or equivalently proxy) as basically equivalent to what net.peer.sock.* used to be.

Reply:
@AlexanderWert does `destination.*` and `source.*` (on client and server side respectively) make sense for these "immediate peers" from ECS perspective?

Reply:
Good question. If we can use destination & source, then maybe we don't even need "proxy"?

Reply:
@trask @Oberon00 

That's a great idea and I think that could be good fit!

In the way that [ECS defines](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html) `source/destination` fields they can co-exist with `client/server` fields:

> Destination fields are usually populated in conjunction with source fields. The source and destination fields are considered the baseline and should always be filled if an event contains source and destination details from a network transaction. If the event also contains identification of the client and server roles, then the client and server fields should also be populated.

Agree with @Oberon00 that we then also need to adapt the existing description of `source/destination` to clarify on the `... when there is no client/server relationship ...` part.
 
Another aspect I'm not sure about is: Are there non-HTTP / non-client-server scenarios with an intermediary? In that case `destination.*` would have a double meaning.

Reply:
> Another aspect I'm not sure about is: Are there non-HTTP / non-client-server scenarios with an intermediary?

Oh good point, there most definitely are. Overloading source/destination could open a can of worms